conbench/tests/api/test_results.py

```diff
@@ -321,167 +319,104 @@ class TestBenchmarkList(_asserts.ListEnforcer):
+                    for run_reason in ["commit", "pr"]:
+                        _fixtures.benchmark_result(
+                            name=name,
+                            batch_id=batch_id,
+                            run_id=run_id,
+                            reason=run_reason,
+                        )
+
+        # ...and another that's too old to be returned without specifying run_id
         _fixtures.benchmark_result(
-            timestamp=(
-                datetime.utcnow().replace(microsecond=0) - timedelta(days=2)
-            ).isoformat()
-            + "Z",
+            name="a", batch_id="3", run_id="1", reason="commit", timestamp="2000-01-01"
         )
-        response = client.get("/api/benchmarks/?days=1")
-        assert len(response.json) == 1
-
-        response = client.get("/api/benchmarks/?days=10")
-        assert len(response.json) == 2
 
-    def test_benchmark_results_with_days_and_limit(self, client):
+    @pytest.fixture(autouse=True)
+    def clear_db_state_between_tests(self):
+        """Override this autoused fixture from conftest.py so we don't delete that fake
+        data between the tests in this class.
+        """
+        return
+
+    # Try all the combinations of filters and different page sizes
+    @pytest.mark.parametrize("run_id_arg", ["1", None])
+    @pytest.mark.parametrize("name_arg", ["a", None])
+    @pytest.mark.parametrize("batch_id_arg", ["3", None])
+    @pytest.mark.parametrize("run_reason_arg", ["commit", None])
+    @pytest.mark.parametrize("page_size_arg", [2, 1000])
+    def test_benchmark_list(
+        self, client, run_id_arg, name_arg, batch_id_arg, run_reason_arg, page_size_arg
+    ):
         self.authenticate(client)
-        _fixtures.benchmark_result()
-        _fixtures.benchmark_result()
-        _fixtures.benchmark_result(
-            timestamp=(
-                datetime.utcnow().replace(microsecond=0) - timedelta(days=2)
-            ).isoformat()
-            + "Z",
-        )
-        response = client.get("/api/benchmarks/?days=10&limit=1")
-        assert len(response.json) == 1
 
-    def test_benchmark_list_filter_by_name_with_daysand_limit(self, client):
-        self.authenticate(client)
-        benchmark_result = _fixtures.benchmark_result(name="bbb")
-        _fixtures.benchmark_result(
-            name="bbb",
-            timestamp=(
-                datetime.utcnow().replace(microsecond=0) - timedelta(days=2)
-            ).isoformat()
-            + "Z",
+        # Find the expected number of filtered results. (Because of the way we set up
+        # the data, each filter should filter out half the remaining results)
+        expected_num_results = 16
+        if name_arg:
+            expected_num_results /= 2
+        if batch_id_arg:
+            expected_num_results /= 2
+        if run_reason_arg:
+            expected_num_results /= 2
+        if run_id_arg:
+            expected_num_results /= 2
```

Comment:
Kind of a pain, but it would be more robust to use a different number of values for each arg here, so a filter applied when it shouldn't be can't hide a missing filter. Not sure whether it's worth the effort, though.

Reply:
Sure I'll change to different prime numbers!